File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
       if(head==null)
       {
           return null;
       }
        ListNode oddptr= head;
        ListNode evenptr=head.next;
        ListNode evenhead= head.next;
        while( evenptr!=null && evenptr.next!=null)
        {
            oddptr.next=oddptr.next.next;
            evenptr.next=evenptr.next.next;
         
            oddptr= oddptr.next;
            evenptr=evenptr.next;
            
        }
        
       oddptr.next=evenhead;
       return head;
        
    }
}

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head;
        oddptr = head;
        evenptr= head.next;
        evenhead = evenptr;
        
        while evenptr and evenptr.next:
            oddptr.next= evenptr.next;
            oddptr = oddptr.next;
            evenptr.next = oddptr.next;
            evenptr= evenptr.next;
        oddptr.next = evenhead;
        return head;
            

            
        